js/genkit/src/client/client.ts

```diff
@@ -15,6 +15,66 @@
  */
 
 import { Channel } from '@genkit-ai/core/async';
+import { z } from '@genkit-ai/core/zod';
+
+interface RunOptions {
+  headers?: Record<string, string>;
+}
+
+export type RemoteAction<I = any, O = any, S = any> = ((
+  input?: I,
+  options?: RunOptions
+) => Promise<O>) & {
+  stream(input?: I, opts?: RunOptions): StreamingResponse<O, S>;
+};
+
+/**
+ * Streaming response from an action.
+ */
+export interface StreamingResponse<O = any, S = any> {
+  /** Iterator over the streaming chunks. */
+  stream: AsyncIterable<S>;
+  /** Final output of the action. */
+  output: Promise<O>;
+}
+
+/**
+ * Defines a remote action which can be invoked as a function or streamed.
+ */
+export function defineRemoteAction<
```

Comment:
should it just be `remoteAction` since there's no registry we're defining it into? but on the other hand `define` feels more consistent with the server side sdk..